### ISSUE TITLE
Swipe is not working Appium 1.6.3 Real device ios

### ISSUE THREAD

**Issue by sudammandhane** [appium/appium#7815]
Swipe is not working 

Running Appium service on terminal[Appium 1.6.3], While performing swipe operation on Appium Inspector[Appium 1.5.3], not at all performing swipe operation, getting 500 error code.

Find logs below 

Last login: Tue Feb  7 11:32:33 on ttys000
GGK-MAC8:~ shivaraj.basodi$ appium
[Appium] Welcome to Appium v1.6.3
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[HTTP] --> GET /wd/hub/status {}
[debug] [MJSONWP] Calling AppiumDriver.getStatus() with args: []
[debug] [MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.6.3","revision":null}}
[HTTP] <-- GET /wd/hub/status 200 20 ms - 83 
[HTTP] --> GET /wd/hub/sessions {}
[debug] [MJSONWP] Calling AppiumDriver.getSessions() with args: []
[debug] [MJSONWP] Responding to client with driver.getSessions() result: []
[HTTP] <-- GET /wd/hub/sessions 200 3 ms - 40 
[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"platformName":"iOS","newCommandTimeout":"7200","app":"/Users/shivaraj.basodi/Downloads/Global Mobile.ipa","platformVersion":"10.0.2","deviceName":"iPhone 6","bundleId":"com.devggktech.PestService","automationName":"Appium","udid":"23a33cfd06dbf1a1a5727838182ec5b1daf82eec"}}
[debug] [MJSONWP] Calling AppiumDriver.createSession() with args: [{"platformName":"iOS","newCommandTimeout":"7200","app":"/Users/shivaraj.basodi/Downloads/Global Mobile.ipa","platformVersion":"10.0.2","deviceName":"iPhone 6","bundleId":"com.devggktech.PestService","automationName":"Appium","udid":"23a33cfd06dbf1a1a5727838182ec5b1daf82eec"},null,null,null,null]
[Appium] Requested iOS support with version >= 10, using XCUITest driver instead of UIAutomation-based driver, since the latter is unsupported on iOS 10 and up.
[Appium] Creating new XCUITestDriver session
[Appium] Capabilities:
[Appium]   platformName: 'iOS'
[Appium]   newCommandTimeout: '7200'
[Appium]   app: '/Users/shivaraj.basodi/Downloads/Global Mobile.ipa'
[Appium]   platformVersion: '10.0.2'
[Appium]   deviceName: 'iPhone 6'
[Appium]   bundleId: 'com.devggktech.PestService'
[Appium]   automationName: 'Appium'
[Appium]   udid: '23a33cfd06dbf1a1a5727838182ec5b1daf82eec'
[debug] [XCUITest] XCUITestDriver version: 2.4.2
[BaseDriver] Capability 'newCommandTimeout' changed from string ('7200') to integer (7200). This may cause unexpected behavior
[BaseDriver] Session created with session id: bbbdd299-04a0-4b31-8ee0-8e563a8e859a
[debug] [XCUITest] Xcode version set to '8.1'
[debug] [XCUITest] iOS SDK Version set to '10.1'
[debug] [XCUITest] Available devices: 23a33cfd06dbf1a1a5727838182ec5b1daf82eec
[debug] [XCUITest] Creating iDevice object with udid 23a33cfd06dbf1a1a5727838182ec5b1daf82eec
[XCUITest] Determining device to run tests on: udid: '23a33cfd06dbf1a1a5727838182ec5b1daf82eec', real device: true
[BaseDriver] Using local app '/Users/shivaraj.basodi/Downloads/Global Mobile.ipa'
[debug] [BaseDriver] Copying local zip to tmp dir
[debug] [BaseDriver] Unzipping /var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/201717-8642-34r78y.hwmfv42t9/appium-app.zip
[debug] [BaseDriver] Testing zip archive: /var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/201717-8642-34r78y.hwmfv42t9/appium-app.zip
[BaseDriver] Unzipped local app to '/var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/201717-8642-34r78y.hwmfv42t9/Payload/Global Mobile.app'
[debug] [XCUITest] Checking whether app '/var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/201717-8642-34r78y.hwmfv42t9/Payload/Global Mobile.app' is actually present
[debug] [XCUITest] App is present
[debug] [iOSLog] Attempting iOS device log capture via libimobiledevice idevicesyslog
[debug] [iOSLog] Found idevicesyslog: '/usr/local/bin/idevicesyslog'
[XCUITest] Setting up real device
[debug] [XCUITest] App is already installed.
[debug] [XCUITest] Full reset not requested. No need to install.
[XCUITest] Using WDA path: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'
[XCUITest] Using WDA agent: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj'
[XCUITest] Launching WebDriverAgent on the device
[debug] [XCUITest] Carthage found: /usr/local/bin/carthage
[debug] [XCUITest] Killing hanging processes
[debug] [XCUITest] Using real device logger 'idevicesyslog'
[debug] [XCUITest] Beginning test with command 'xcodebuild build test -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=23a33cfd06dbf1a1a5727838182ec5b1daf82eec -configuration Debug' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'
[debug] [XCUITest] Unable to get device time. Using local system time
[debug] [XCUITest] Output from idevicedate: Tue Feb  7 11:55:16 IST 2017

[XCUITest] Waiting for WebDriverAgent to start on device
[debug] [XCUITest] Log file for xcodebuild test: /Users/shivaraj.basodi/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/A0FC1198-6E1A-4E23-95B4-6FF822957AEC/Session-WebDriverAgentRunner-2017-02-07_115517-RLa1sV.log
[debug] [WebDriverAgent] Device: Feb  7 11:55:26 iPhone XCTRunner[3286] <Notice>: Running tests...
[debug] [WebDriverAgent] Device: Feb  7 11:55:28 iPhone XCTRunner[3286] <Notice>: Continuing to run tests in the background with task ID 1
[XCUITest] Detected that WebDriverAgent is running at url 'http://172.16.7.97:8100'
[XCUITest] WebDriverAgent started at url 'http://172.16.7.97:8100'
[debug] [XCUITest] Starting iproxy to forward traffic from local port 8100 to device port 8100 over USB
[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}
[debug] [JSONWP Proxy] Got response with status 200: {"value":{"sessionId":"7BB7E3D5-4799-4093-97DF-D12D3A591463","capabilities":{"device":"iphone","browserName":"PestService","sdkVersion":"10.0.2","CFBundleIdentifier":"com.devggktech.PestService"}},"sessionId":"7BB7E3D5-4799-4093-97DF-D12D3A591463","status":0}
[debug] [XCUITest] Setting initial orientation to 'PORTRAIT'
[debug] [JSONWP Proxy] Proxying [POST /orientation] to [POST http://localhost:8100/session/7BB7E3D5-4799-4093-97DF-D12D3A591463/orientation] with body: {"orientation":"PORTRAIT"}
[debug] [JSONWP Proxy] Got response with status 200: {"value":{},"sessionId":"7BB7E3D5-4799-4093-97DF-D12D3A591463","status":0}
[Appium] New XCUITestDriver session created successfully, session bbbdd299-04a0-4b31-8ee0-8e563a8e859a added to master session list
[debug] [MJSONWP] Responding to client with driver.createSession() result: {"webStorageEnabled":false,"locationContextEnabled":false,"browserName":"","platform":"MAC","javascriptEnabled":true,"databaseEnabled":false,"takesScreenshot":true,"networkConnectionEnabled":false,"platformName":"iOS","newCommandTimeout":"7200","app":"/Users/shivaraj.basodi/Downloads/Global Mobile.ipa","platformVersion":"10.0.2","deviceName":"iPhone 6","bundleId":"com.devggktech.PestService","automationName":"Appium","udid":"23a33cfd06dbf1a1a5727838182ec5b1daf82eec"}
[HTTP] <-- POST /wd/hub/session 200 22564 ms - 543 
[HTTP] --> GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/source {}
[debug] [MJSONWP] Calling AppiumDriver.getPageSource() with args: ["bbbdd299-04a0-4b31-8ee0-8e563a8e859a"]
[debug] [XCUITest] Executing command 'getPageSource'
[debug] [JSONWP Proxy] Proxying [POST /source] to [POST http://localhost:8100/session/7BB7E3D5-4799-4093-97DF-D12D3A591463/source] with no body
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"tree\" : {\n      \"isEnabled\" : \"1\",\n      \"isVisible\" : \"1\",\n      \"frame\" : \"{{0, 0}, {375, 667}}\",\n      \"children\" : [\n        {\n          \"isEnabled\" : \"1\",\n          \"isVisible\" : \"1\",\n          \"frame\" : \"{{0, 0}, {375, 667}}\",\n          \"children\" : [\n            {\n              \"isEnabled\" : \"1\",\n              \"isVisible\" : \"1\",\n              \"frame\" : \"{{0, 0}, {375, 667}}\",\n              \"children\" : [\n                {\n                  \"isEnabled\" : \"1\",\n                  \"isVisible\" : \"1\",\n                  \"frame\" : \"{{0, 0}, {375, 667}}\",\n                  \"children\" : [\n                    {\n                      \"isEnabled\" : \"1\",\n                      \"isVisible\" : \"1\",\n                      \"frame\" : \"{{0, 0}, {375, 667}}\",\n                      \"children\" : [\n                        {\n                          \"isEnabled\" : \"1\",\n                          \"isVi...
[debug] [MJSONWP] Responding to client with driver.getPageSource() result: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<AppiumAUT>\n    <XCUIElementTypeApplication name=\"PestService\" label=\"PestService\" value=\"\" dom=\"\" enabled=\"true\" valid=\"true\" visible=\"true\" hint=\"\" path=\"/0\" x=\"0\" y=\"0\" width=\"375\" height=\"667\">\n        <XCUIElementTypeWindow name=\"\" label=\"\" value=\"\" dom=\"\" enabled=\"true\" valid=\"true\" visible=\"true\" hint=\"\" path=\"/0/0\" x=\"0\" y=\"0\" width=\"375\" height=\"667\">\n            <XCUIElementTypeOther name=\"\" label=\"\" value=\"\" dom=\"\" enabled=\"true\" valid=\"true\" visible=\"true\" hint=\"\" path=\"/0/0/0\" x=\"0\" y=\"0\" width=\"375\" height=\"667\">\n                <XCUIElementTypeOther name=\"\" label=\"\" value=\"\" dom=\"\" enabled=\"true\" valid=\"true\" visible=\"true\" hint=\"\" path=\"/0/0/0/0\" x=\"0\" y=\"0\" width=\"375\" height=\"667\">\n                    <XCUIElementTypeOther name=\"\" label=\"\" value=\"\" dom=\"\" enabled=\"true\" valid=\"true\" visible=\"true\" hint=\"\" path=\"/0/0/0/0/...
[HTTP] <-- GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/source 200 855 ms - 17569 
[HTTP] --> GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/orientation {}
[MJSONWP] Driver proxy active, passing request on via HTTP proxy
[debug] [XCUITest] Executing command 'proxyReqRes'
[debug] [JSONWP Proxy] Proxying [GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/orientation] to [GET http://localhost:8100/session/7BB7E3D5-4799-4093-97DF-D12D3A591463/orientation] with body: {}
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : \"PORTRAIT\",\n  \"sessionId\" : \"7BB7E3D5-4799-4093-97DF-D12D3A591463\",\n  \"status\" : 0\n}"
[JSONWP Proxy] Replacing sessionId 7BB7E3D5-4799-4093-97DF-D12D3A591463 with bbbdd299-04a0-4b31-8ee0-8e563a8e859a
[HTTP] <-- GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/orientation 200 159 ms - 82 
[HTTP] --> GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/screenshot {}
[MJSONWP] Driver proxy active, passing request on via HTTP proxy
[debug] [XCUITest] Executing command 'proxyReqRes'
[debug] [JSONWP Proxy] Proxying [GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/screenshot] to [GET http://localhost:8100/session/7BB7E3D5-4799-4093-97DF-D12D3A591463/screenshot] with body: {}
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : \"iVBORw0KGgoAAAANSUhEUgAAAu4AAAU2CAYAAADK1zMGAAAAAXNSR0IArs4c6QAA\\r\\nABxpRE9UAAAAAgAAAAAAAAKbAAAAKAAAApsAAAKbAADMEYxamX0AAEAASURBVHgB\\r\\n7J0HvNzEucXBDr23BBJCDYQWCC3thUBIKKkvyaNjDLbpvfcOpptuCBgwvZje4brh\\r\\ngm3cwBVjiummYwymw7z5a+\\/sndVKu9p6V7tnfr+92quVppw5ks58+uab+eY70Rh9\\r\\nhIE4IA6IA+KAOCAOiAPigDjQ4BxQBzV4B2lgpYGlOCAOiAPigDggDogD4gAckHCX\\r\\ncBcHxAFxQBwQB8QBcUAcEAdSwAF1Ugo6SaNsjbLFAXFAHBAHxAFxQBwQByTcJdzF\\r\\nAXGgZA6c8L2Z\\/+i5Zv7D3jZdDnzJdNl3qunSa4Lp0uMZ02Wvp03X7sNN1z2Gmq7d\\r\\nBpuuuw+yn4H209akH9pm20hbabNtOxgEWIAJ2FiMwArM5rPYlYy3HtbCTBwQB8QB\\r\\ncQAO6AEi0SYOiAOJOHDMPDP\\/QbNMl57jrEhFjDerEK9xuyx2YAiW81lME2GvB7Zw\\r\\nEgfEAXFAHJBwl2CTaBAHCnIAyzpWdWtJl1CvjaAHWzCWJV7XYsFrUaJNok0cEAck\\r\\n3PWg0INCHIjjAGIycHmRZb0+g5Y9R2QEvB7OejiLA+KAOCAOxHEg7qGt\\/RJ04kCL\\r\\ncuCYz2Rh78TBChb4+Wwf6Ppr0esv7mGt\\/bomxAFxAA7o4aCHgzggDjgOzH\\/I65mJ\\r\\nlp0oXOWSg0vOIENfuH7RVteoOCAOiAPiQMABEUFEEAfEAfyriX4i0VwbP\\/ZycaVP\\r\\n5Puu61P3aHFAHGg+DmzVr8w2NTsZljrTmE2uNObP1xuzxkXGdD25TKAa7BXVAqcY\\r\\ns21\\/Y...
[JSONWP Proxy] Replacing sessionId 7BB7E3D5-4799-4093-97DF-D12D3A591463 with bbbdd299-04a0-4b31-8ee0-8e563a8e859a
[HTTP] <-- GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/screenshot 200 78 ms - 147866 
[HTTP] --> GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/contexts {}
[debug] [MJSONWP] Calling AppiumDriver.getContexts() with args: ["bbbdd299-04a0-4b31-8ee0-8e563a8e859a"]
[debug] [XCUITest] Executing command 'getContexts'
[debug] [iOS] Getting list of available contexts
[debug] [iOS] Retrieving contexts and views
[debug] [iOS] Selecting by url: false 
[debug] [RemoteDebugger] Getting WebKitRemoteDebugger pageArray: localhost, 27753
[debug] [RemoteDebugger] Sending request to: http://localhost:27753/json
[iOS] Attempted to get a list of webview contexts but could not connect to ios-webkit-debug-proxy. If you expect to find webviews, please ensure that the proxy is running and accessible
[debug] [iOS] No web frames found.
[debug] [MJSONWP] Responding to client with driver.getContexts() result: ["NATIVE_APP"]
[HTTP] <-- GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/contexts 200 28 ms - 86 
[HTTP] --> POST /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/execute {"script":"mobile: swipe","args":[{"endX":179,"duration":2,"endY":180,"touchCount":1,"startX":173,"startY":548}]}
[debug] [MJSONWP] Calling AppiumDriver.execute() with args: ["mobile: swipe",[{"endX":179,"duration":2,"endY":180,"touchCount":1,"startX":173,"startY":548}],"bbbdd299-04a0-4b31-8ee0-8e563a8e859a"]
[debug] [XCUITest] Executing command 'execute'
[HTTP] <-- POST /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/execute 500 19 ms - 159 
[HTTP] --> GET /wd/hub/session/bbbdd299-04a0-4b31-8ee0-8e563a8e859a/source {}

**Comment by mykola-mokhnach** [comment]
mobile:swipe is not supported in 1.6.3
Use need to use TouchAction interface for this purpose now.

**Comment by iceKirthi** [comment]
@mykola-mokhnach : Could you please share the code to swipe using TouchAction ?

**Comment by imurchie** [comment]
`mobile: swipe` is available in 1.6.4-beta for XCUITest-based iOS tests.

**Comment by dilsreccse** [comment]
Tried swipe using 1.6.4 beta version, in that swipe from bottom to top is not working, [Gist Here](https://gist.github.com/dilsreccse/a9682d833be0bad968852caa5dabc2cc) . 

@imurchie & @mykola-mokhnach need support with the swipe issues

**Comment by bgalperin** [comment]
I am running into the same issue of not being able to swipe on a real iOS device though oddly enough It works on the simulator.

I tried using Appium 1.6.4-beta, but the issue still occurs.

Here is a gist using AppiumDriver.swipe(): [Swipe](https://gist.github.com/bgalperin/52e58ac0c38a93d566b9198a2b86751a)
Here is a gist using a JavascriptExecutor to scroll down: [scroll down](https://gist.github.com/bgalperin/e6cb253fcadb0559c8eb347e0578bc5e)

If this turns out to be a separate issue, let me know I can open up a different issue

**Comment by bgalperin** [comment]
Using `mobile: swipe`, `mobile: scroll`, or `AppiumDriver.swipe()` all fail on a real iOS 10 device with the `Unhandled endpoint` error. 
A sample appium log is [here](https://gist.github.com/bgalperin/e6cb253fcadb0559c8eb347e0578bc5e)
 
Any ideas?

**Comment by eyypbgds** [comment]
@sudammandhane @bgalperin Please try this;

WebElement element = driver.findElementByClassName("XCUIElementTypeApplication");
int x = element.getSize().getWidth();
int xEnd = 0;
int yStart = element.getSize().getHeight()/2;
driver.swipe(x, yStart, xEnd, yStart, 500);

**Comment by bgalperin** [comment]
@eyypbgds I have already tried that method, that is what I referred to when I said `AppiumDriver.swipe()`

I am using 
`driver.swipe(startX, startY, endX, endY, 1000);`
Where `startX`,`startY`,`endX`,`andY` are all integers I am feeding in. 
In the case of the log below, you can see they are 75, 533, 75, 133.

e.g.
`[MJSONWP] Calling AppiumDriver.performTouch() with args: [[{"action":"press","options":{"x":75,"y":533}},{"action":"wait","options":{"ms":1000}},{"action":"moveTo","options":{"x":0,"y":-400}},{"action":"release","options":{}}],"8eef5a17-d7fa-450b-b8e4-aa14db8699cc"]
[debug] [XCUITest] Executing command 'performTouch'
[debug] [XCUITest] Received the following touch action: press-wait-moveTo-release
debug] [JSONWP Proxy] Proxying [POST /wda/element/0/dragfromtoforduration] to [POST http://localhost:8100/session/88AE084A-744D-49D3-9283-C5CF98AFB6F9/wda/element/0/dragfromtoforduration] with body: {"fromX":75,"fromY":533,"toX":75,"toY":133,"duration":1}
[debug] [JSONWP Proxy] Got response with status 200: {"value":"Unhandled endpoint: /session/88AE084A-744D-49D3-9283-C5CF98AFB6F9/wda/element/0/dragfromtoforduration -- http://localhost:8100/ with parameters {\n    wildcards =     (\n        \"session/88AE084A-744D-49D3-9283-C5CF98AFB6F9/wda/element/0/dragfromtoforduration\"\n    );\n}","sessionId":"88AE084A-744D-49D3-9283-C5CF98AFB6F9","status":1}`

**Comment by mykola-mokhnach** [comment]
@bgalperin Please install the recent Appium 1.6.4-beta containing appium-xcuitest-driver v. 2.10.0 or higher and try to reproduce the issue

**Comment by bgalperin** [comment]
@mykola-mokhnach I made sure to install the latest Appium 1.6.4-beta, and confirmed appium-xcuitest-driver was on version 2.10.0, and still get the same 'Unhandled endpoint' error when using a real iOS 10 device.

My latest appium server logs from the run with 'driver.swipe()': [driver.swipe](https://gist.github.com/bgalperin/80220534f5ce056fb6fd66513aefbc77)

And with 'mobile: swipe': [mobile: swipe](https://gist.github.com/bgalperin/e0ac7d9218f21abed19057f137658e63)

Just in case, I tried the 'mobile: swipe' code on a simulator, and it passed. For reference, that log is here: [simulator mobile: swipe](https://gist.github.com/bgalperin/56f70cc560b24c8bc6bc6b5abd82afaf)

**Comment by mykola-mokhnach** [comment]
@bgalperin Please make sure the WebDriverAgent is updated on your device first. You may uninstall it manually from there and then let Appium to compile and install it again (WDA application has a simple white icon)

**Comment by bgalperin** [comment]
@mykola-mokhnach I uninstalled it manually, ran the test with 'mobile: swipe' and received the same 'Unhandled endpoint' error I received in my previous post. 
Are there any other logs / info I can provide to help debug?

**Comment by mykola-mokhnach** [comment]
can you please attach the full log again?

**Comment by bgalperin** [comment]
Sure, this is from the last run on a real device, with manually uninstalling the WDA app, and using 'mobile: swipe': [gist](https://gist.github.com/bgalperin/e907d89fa155709ad4b381d8b3004edb)

**Comment by mykola-mokhnach** [comment]
This is super strange. Are you sure you have the latest WDA source?
Do you observe the same output if you grep the folder where appium-xcuitest-driver component is installed?

```bash
$ grep -R '/wda/element/:uuid/swipe' *
WebDriverAgent/WebDriverAgentLib/Commands/FBElementCommands.m:    [[FBRoute POST:@"/wda/element/:uuid/swipe"] respondWithTarget:self action:@selector(handleSwipe:)],
```

**Comment by bgalperin** [comment]
My output was:
`WebDriverAgent/WebDriverAgentLib/Commands/FBElementCommands.m:    [[FBRoute POST:@"/wda/element/:uuid/swipe"] respondWithTarget:self action:@selector(handleSwipe:)],`

**Comment by mykola-mokhnach** [comment]
@imurchie Can you please check? I'm out of ideas here.

**Comment by sudammandhane** [comment]
How to install Appin 1.6.4 beta?

**Comment by vikramvi** [comment]
> @bgalperin Please install the recent Appium 1.6.4-beta containing appium-xcuitest-driver v. 2.10.0 or higher and try to reproduce the issue

@mykola-mokhnach  yesterday I've installed `appium 1.6.4-beta `and when I checked version of appium-xcunitest-driver with command `npm list -g`, it's showing `appium-xcuitest-driver@2.9.0`

Is this bug with latest version of appium, which is pointing to older version of xcuitest driver ?

**Comment by vikramvi** [comment]
> How to install Appin 1.6.4 beta?

@sudammandhane https://discuss.appium.io/t/appium-1-6-4-beta-released/14543

**Comment by mykola-mokhnach** [comment]
@vikramvi Appium beta does have component versions strictly defined, which means it automatically receives the most recent (or satisfying restrictions from _package.json_ to be more precise) component versions available as soon as you reinstall it.

**Comment by vikramvi** [comment]
@mykola-mokhnach Thanks, please clarify how to remove existing version of appium-xcuitest-driver and install a new one ?

**Comment by mykola-mokhnach** [comment]
just reinstall the whole Appium:

```
npm remove -g appium
npm install -g appium@1.6.4-beta
```

**Comment by bgalperin** [comment]
@mykola-mokhnach I cleaned the WebDriverAgent project in appium 1.6.4-beta, and I can scroll / swipe on the real iOS device! 
Thank you for your time

**Comment by qabotnet** [comment]
@bgalperin Did you do any other step than :
`npm remove -g appium
npm install -g appium@1.6.4-beta` ?

I did those command and removed WebdriverAgent for my IPhone but  i am unable to do touch action. Maybe because i am using IOS 9.3.

**Comment by coolzire** [comment]
@bgalperin 
Did you do any other step?
I removed the webdriveragent from my device then used:
`npm remove -g appium
npm install -g appium@1.6.4-beta`

But touch action don't work as expected. Maybe it because I am using a Iphone on IOS 9.3?

**Comment by bgalperin** [comment]
@coolzire I performed the step you mentioned (removing currently installed appium, then installing 1.6.4 beta), then I opened the WebDriverAgent project in Xcode and cleaned it, and made sure it was properly signed by my development team, then rebuilt it.

I am using an iOS 10.2.1 device

**Comment by coolzire** [comment]
@bgalperin  I managed to track down the issue I had. TouchAction uses Point not Pixel coordinates.
I am not a mobile developper thus was unaware of this ios feature.

**Comment by vikramvi** [comment]
@bgalperin can you please share methods how you achieved scroll and swipe.

As I'm still struck blocked because of  https://github.com/appium/appium/issues/7914

Thanks in advance.

**Comment by bgalperin** [comment]
I am using both swipe and scroll, in the formats:
(Where "driver" is AppiumDriver<MobileElement>)
Swipe:
`driver.swipe(startX, startY, endX, endY, duration);`

And scroll, in this case down:
`        JavascriptExecutor js = (JavascriptExecutor) driver;
        HashMap<String, String> scrollObject = new HashMap<String, String>();
        scrollObject.put("direction", "down");
        js.executeScript("mobile: scroll", scrollObject);`

**Comment by Dheepthi** [comment]
@bgalperin how to achieve mobile:swipe with co-ordinates?

JavascriptExecutor js = (JavascriptExecutor) driver;
HashMap<String, String> swipeObject = new HashMap<String, String>();
swipeObject.put("direction", "right");

like this driver.swipe(left, height, right, height, swipeDuration);

**Comment by oscheller1** [comment]
Can confirm that `driver.swipe` is working in Appium 1.6.4beta. `TouchAction` isn't working though. Any updates?

**Comment by coolzire** [comment]
Select TouchAction methods are working in 1.6.4 beta.

**Comment by bgalperin** [comment]
@Dheepthi I unfortunately don't know. I was using coordinates with driver.swipe, and just directions with the JavascriptExecutor code.

**Comment by sudammandhane** [comment]
Ran following commands 
npm remove -g appium
npm install -g appium@1.6.4-beta

Now getting "@#[line:0,col:undefined] [xmldom error]	element parse error: TypeError: source.indexOf is not a function" error. Please find following logs 

GGK-MAC8:~ shivaraj.basodi$ appium

[Appium] Welcome to Appium v1.6.4-beta

[Appium] Appium REST http interface listener started on 0.0.0.0:4723

[HTTP] --> GET /wd/hub/status {}

[debug] [MJSONWP] Calling AppiumDriver.getStatus() with args: []

[debug] [MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.6.4-beta","revision":null}}

[HTTP] <-- GET /wd/hub/status 200 20 ms - 88 

[HTTP] --> GET /wd/hub/sessions {}

[debug] [MJSONWP] Calling AppiumDriver.getSessions() with args: []

[debug] [MJSONWP] Responding to client with driver.getSessions() result: []

[HTTP] <-- GET /wd/hub/sessions 200 5 ms - 40 

[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"platformName":"iOS","newCommandTimeout":"7200","app":"/Users/shivaraj.basodi/Desktop/Global Mobile 2017-03-20 14-54-49/Global Mobile.ipa","platformVersion":"10.0.2","deviceName":"iPhone 6","bundleId":"com.devggktech.PestService","automationName":"Appium","udid":"23a33cfd06dbf1a1a5727838182ec5b1daf82eec"}}

[debug] [MJSONWP] Calling AppiumDriver.createSession() with args: [{"platformName":"iOS","newCommandTimeout":"7200","app":"/Users/shivaraj.basodi/Desktop/Global Mobile 2017-03-20 14-54-49/Global Mobile.ipa","platformVersion":"10.0.2","deviceName":"iPhone 6","bundleId":"com.devggktech.PestService","automationName":"Appium","udid":"23a33cfd06dbf1a1a5727838182ec5b1daf82eec"},null,null,null,null]

[debug] [BaseDriver] Event 'newSessionRequested' logged at 1490006772318 (16:16:12 GMT+0530 (IST))

[Appium] Requested iOS support with version >= 10, using XCUITest driver instead of UIAutomation-based driver, since the latter is unsupported on iOS 10 and up.

[Appium] Creating new XCUITestDriver (v2.22.0) session

[Appium] Capabilities:

[Appium]   platformName: 'iOS'

[Appium]   newCommandTimeout: '7200'

[Appium]   app: '/Users/shivaraj.basodi/Desktop/Global Mobile 2017-03-20 14-54-49/Global Mobile.ipa'

[Appium]   platformVersion: '10.0.2'

[Appium]   deviceName: 'iPhone 6'

[Appium]   bundleId: 'com.devggktech.PestService'

[Appium]   automationName: 'Appium'

[Appium]   udid: '23a33cfd06dbf1a1a5727838182ec5b1daf82eec'

[debug] [XCUITest] XCUITestDriver version: 2.22.0

[BaseDriver] Capability 'newCommandTimeout' changed from string ('7200') to integer (7200). This may cause unexpected behavior

[BaseDriver] Session created with session id: fc5612aa-5cf9-45ed-bace-24a33c4e3506

[debug] [XCUITest] Xcode version set to '8.1' (tools v8.2.0.0.1.1480973914)

[debug] [XCUITest] iOS SDK Version set to '10.1'

[debug] [BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1490006772683 (16:16:12 GMT+0530 (IST))

[debug] [XCUITest] Available devices: 23a33cfd06dbf1a1a5727838182ec5b1daf82eec

[debug] [XCUITest] Creating iDevice object with udid '23a33cfd06dbf1a1a5727838182ec5b1daf82eec'

[XCUITest] Determining device to run tests on: udid: '23a33cfd06dbf1a1a5727838182ec5b1daf82eec', real device: true

[BaseDriver] Using local app '/Users/shivaraj.basodi/Desktop/Global Mobile 2017-03-20 14-54-49/Global Mobile.ipa'

[debug] [BaseDriver] Copying local zip to tmp dir

[debug] [BaseDriver] Unzipping /var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/2017220-46122-1eq16h0.nsy9fi529/appium-app.zip

[debug] [BaseDriver] Testing zip archive: /var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/2017220-46122-1eq16h0.nsy9fi529/appium-app.zip

[BaseDriver] Unzipped local app to '/var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/2017220-46122-1eq16h0.nsy9fi529/Payload/Global Mobile.app'

[debug] [BaseDriver] Event 'appConfigured' logged at 1490006774292 (16:16:14 GMT+0530 (IST))

[debug] [XCUITest] Checking whether app '/var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/2017220-46122-1eq16h0.nsy9fi529/Payload/Global Mobile.app' is actually present on file system

[debug] [XCUITest] App is present

[debug] [BaseDriver] Event 'resetStarted' logged at 1490006774295 (16:16:14 GMT+0530 (IST))

[debug] [XCUITest] Reset: running ios real device reset flow

[debug] [BaseDriver] Event 'resetComplete' logged at 1490006774297 (16:16:14 GMT+0530 (IST))

[debug] [iOSLog] Attempting iOS device log capture via libimobiledevice idevicesyslog

[debug] [iOSLog] Found idevicesyslog: '/usr/local/bin/idevicesyslog'

[debug] [BaseDriver] Event 'logCaptureStarted' logged at 1490006774494 (16:16:14 GMT+0530 (IST))

[XCUITest] Setting up real device

[debug] [XCUITest] Reset requested. Removing app with id 'com.devggktech.PestService' from the device

[debug] [XCUITest] Installing '/var/folders/yd/by8pgtjj6918hx6vs89vhczdsngs2s/T/2017220-46122-1eq16h0.nsy9fi529/Payload/Global Mobile.app' on device with UUID '23a33cfd06dbf1a1a5727838182ec5b1daf82eec'...

[debug] [XCUITest] The app has been installed successfully.

[debug] [BaseDriver] Event 'appInstalled' logged at 1490006783022 (16:16:23 GMT+0530 (IST))

[debug] [BaseDriver] Event 'wdaStartAttempted' logged at 1490006783025 (16:16:23 GMT+0530 (IST))

[XCUITest] Using WDA path: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'

[XCUITest] Using WDA agent: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj'

[XCUITest] Launching WebDriverAgent on the device

[debug] [XCUITest] Carthage found: '/usr/local/bin/carthage'

[debug] [XCUITest] Killing hanging processes

[debug] [XCUITest] Beginning test with command 'xcodebuild build-for-testing test-without-building -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=23a33cfd06dbf1a1a5727838182ec5b1daf82eec -configuration Debug' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'

[debug] [XCUITest] Output from xcodebuild will not be logged

[debug] [XCUITest] Starting iproxy to forward traffic from local port 8100 to device port 8100 over USB

[debug] [XCUITest] Waiting up to 60000ms for WebDriverAgent to start

[debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"state\" : \"success\",\n    \"os\" : {\n      \"name\" : \"iOS\",\n      \"version\" : \"10.0.2\"\n    },\n    \"ios\" : {\n      \"simulatorVersion\" : \"10.0.2\",\n      \"ip\" : \"172.16.7.97\"\n    },\n    \"build\" : {\n      \"time\" : \"Jan 25 2017 17:05:28\"\n    }\n  },\n  \"sessionId\" : \"E994B362-9C1A-48A3-86E3-30D7FA1D2B99\",\n  \"status\" : 0\n}"

[debug] [XCUITest] WebDriverAgent successfully started after 3716ms

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006791927 (16:16:31 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Log file for xcodebuild test: /Users/shivaraj.basodi/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/C17B2C68-56C7-4519-B2EE-9767953CDB3D/Session-WebDriverAgentRunner-2017-03-20_161633-CO5u5n.log

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006797461 (16:16:37 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006798475 (16:16:38 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006799489 (16:16:39 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006800503 (16:16:40 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006801517 (16:16:41 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006802535 (16:16:42 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006803549 (16:16:43 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006804561 (16:16:44 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [XCUITest] Failed to create WDA session. Retrying...

[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1490006805578 (16:16:45 GMT+0530 (IST))

[debug] [XCUITest] Sending createSession command to WDA

[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.devggktech.PestService","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

[debug] [JSONWP Proxy] Got response with status 200: {"value":{"sessionId":"9F40C2D4-A621-4B27-808B-1E8358EF0740","capabilities":{"device":"iphone","browserName":"PestService","sdkVersion":"10.0.2","CFBundleIdentifier":"com.devggktech.PestService"}},"sessionId":"9F40C2D4-A621-4B27-808B-1E8358EF0740","status":0}

[debug] [BaseDriver] Event 'wdaSessionStarted' logged at 1490006809292 (16:16:49 GMT+0530 (IST))

[debug] [XCUITest] Found WDA derived data folder: '/Users/shivaraj.basodi/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy'

[XCUITest] Setting '555' permissions to '/Users/shivaraj.basodi/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/Attachments' folder

[debug] [XCUITest] Found WDA derived data folder: '/Users/shivaraj.basodi/Library/Developer/Xcode/DerivedData/WebDriverAgent-dfbwkfljszxhtvfzqoiaqfvegsnw'

[XCUITest] Setting '555' permissions to '/Users/shivaraj.basodi/Library/Developer/Xcode/DerivedData/WebDriverAgent-dfbwkfljszxhtvfzqoiaqfvegsnw/Logs/Test/Attachments' folder

[debug] [BaseDriver] Event 'wdaPermsAdjusted' logged at 1490006809309 (16:16:49 GMT+0530 (IST))

[debug] [BaseDriver] Event 'wdaStarted' logged at 1490006809310 (16:16:49 GMT+0530 (IST))

[debug] [XCUITest] Setting initial orientation to 'PORTRAIT'

[debug] [JSONWP Proxy] Proxying [POST /orientation] to [POST http://localhost:8100/session/9F40C2D4-A621-4B27-808B-1E8358EF0740/orientation] with body: {"orientation":"PORTRAIT"}

[debug] [JSONWP Proxy] Got response with status 200: {"value":{},"sessionId":"9F40C2D4-A621-4B27-808B-1E8358EF0740","status":0}

[debug] [BaseDriver] Event 'orientationSet' logged at 1490006809583 (16:16:49 GMT+0530 (IST))

[Appium] New XCUITestDriver session created successfully, session fc5612aa-5cf9-45ed-bace-24a33c4e3506 added to master session list

[debug] [BaseDriver] Event 'newSessionStarted' logged at 1490006809586 (16:16:49 GMT+0530 (IST))

[debug] [MJSONWP] Responding to client with driver.createSession() result: {"webStorageEnabled":false,"locationContextEnabled":false,"browserName":"","platform":"MAC","javascriptEnabled":true,"databaseEnabled":false,"takesScreenshot":true,"networkConnectionEnabled":false,"platformName":"iOS","newCommandTimeout":"7200","app":"/Users/shivaraj.basodi/Desktop/Global Mobile 2017-03-20 14-54-49/Global Mobile.ipa","platformVersion":"10.0.2","deviceName":"iPhone 6","bundleId":"com.devggktech.PestService","automationName":"Appium","udid":"23a33cfd06dbf1a1a5727838182ec5b1daf82eec"}

[HTTP] <-- POST /wd/hub/session 200 37272 ms - 575 

[HTTP] --> GET /wd/hub/session/fc5612aa-5cf9-45ed-bace-24a33c4e3506/source {}

[debug] [MJSONWP] Calling AppiumDriver.getPageSource() with args: ["fc5612aa-5cf9-45ed-bace-24a33c4e3506"]

[debug] [XCUITest] Executing command 'getPageSource'

[debug] [JSONWP Proxy] Proxying [GET /source] to [GET http://localhost:8100/session/9F40C2D4-A621-4B27-808B-1E8358EF0740/source] with no body

[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"tree\" : {\n      \"isEnabled\" : \"1\",\n      \"isVisible\" : \"1\",\n      \"frame\" : \"{{0, 0}, {375, 667}}\",\n      \"children\" : [\n        {\n          \"isEnabled\" : \"1\",\n          \"isVisible\" : \"1\",\n          \"frame\" : \"{{0, 0}, {375, 667}}\",\n          \"children\" : [\n            {\n              \"isEnabled\" : \"1\",\n              \"isVisible\" : \"1\",\n              \"frame\" : \"{{0, 0}, {375, 667}}\",\n              \"children\" : [\n                {\n                  \"isEnabled\" : \"1\",\n                  \"isVisible\" : \"1\",\n                  \"frame\" : \"{{0, 0}, {375, 667}}\",\n                  \"children\" : [\n                    {\n                      \"isEnabled\" : \"1\",\n                      \"isVisible\" : \"1\",\n                      \"frame\" : \"{{0, 0}, {375, 667}}\",\n                      \"children\" : [\n                        {\n                          \"isEnabled\" : \"1\",\n                          \"isVi...

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function 

@#[line:0,col:undefined]

[xmldom error]	element parse error: TypeError: source.indexOf is not a function

**Comment by mykola-mokhnach** [comment]
@sudammandhane your issue is different. It looks like you still have some WDA leftovers after Appium leftovers.

Try to remove WebDriverAgent application from your device under test, reboot your computer to make sure no files are locked in Appium folder and only after that update Appium to beta.

**Comment by sjenasears** [comment]
driver.swipe(400, 50, -900, 50, 1000);
Not working for simulator in latest appim 1.6.4 beta

**Comment by bgalperin** [comment]
@sjenasears I believe the driver.swipe(startX, startY, endX, endY, duration) command does not accept negative values, only positive ones where the coordinates 0,0 corresponds to the top left of the screen

**Comment by vjmaiky** [comment]
@bgalperin can you please try to use your swipe in order to Bring iOS control center? i tried but seems not to work..

**Comment by pawankums** [comment]
@mykola-mokhnach @bgalperin 
Guys i need to scroll up/down to an element which is not in view in IOS simulator. Could you please let me know the code to do so. I am new to  it. I have installed the appium 1.6.4 beta as well.

**Comment by mykola-mokhnach** [comment]
@pawankums Please use Appium forum to ask questions. This is the place where we solve issues.

**Comment by PrabhatPandey** [comment]
Apparently in java-client-5.0.0-beta6, I was using 4.1.2 This bottom to top swipe issue is resolved. I am using 1.6.3 Appium , Xcode is 8.

Putting the same code which work in iPhone-5s-10.2

public static void SwitchWifiBar(AppiumDriver driver) {
final int height = driver.findElementByClassName("UIAWindow").getSize().getHeight();
final int width = driver.findElementByClassName("UIAWindow").getSize().getWidth();
System.out.println("height"+height);
System.out.println("width"+width);
driver.swipe(width-100, height-5, width-100,0, 100);
driver.findElementByAccessibilityId("Wi-Fi").click();
}

**Comment by vikrammobile** [comment]
For me the Below script worked with Appium 1.6.5

  javascriptExecutor js = (JavascriptExecutor) driver;
		HashMap<String, String> scrollObject = new HashMap<String, String>();
		scrollObject.put("direction", "left");
		//scrollObject.put("element", ((IOSElement) ELEMENT).getId());
		js.executeScript("mobile: swipe", scrollObject);
